### PR TITLE
Note which primitives can currently be batched

### PR DESCRIPTION
**Pull request by lawnjelly**
This is only relevant to 3.x branch, 3.3 onward.

This is my first attempt at a PR to the docs like this so may go horribly wrong...

I'd also ideally like to put a link into:
https://docs.godotengine.org/en/3.3/tutorials/2d/custom_drawing_in_2d.html

But I don't know how to.

Fixes https://github.com/godotengine/godot/issues/50387

### CODE REVIEW

File: tutorials/optimization/batching.rst [hunk 1]
```diff
@@ -514,6 +514,21 @@ I use a large number of textures, so few items are being batched.
...
+especially with primitives using an anti-aliased border. The following
+primitive types are currently available:
+
+- RECT
+- NINEPATCH (depending on wrapping mode)
+- POLY
+- LINE
+
+With non-batched primitives you may be able to get better performance drawing
+them manually with polys in a ``_draw()`` function.
```
Comment (Calinou) [inline tutorials/optimization/batching.rst:530]:
```suggestion
Batched primitives
^^^^^^^^^^^^^^^^^^

Not all primitives can be batched. Batching is not guaranteed either,
especially with primitives using an antialiased border. The following
primitive types are currently available:

- RECT
- NINEPATCH (depending on wrapping mode)
- POLY
- LINE

With non-batched primitives, you may be able to get better performance by
drawing them manually with polys in a ``_draw()`` function.
See :ref:`doc_custom_drawing_in_2d` for more information.
```

**Comment by lawnjelly** [comment]
Thanks for changes @Calinou ! :smile:

**Comment by akien-mga** [comment]
> I'd also ideally like to put a link into:
docs.godotengine.org/en/3.3/tutorials/2d/custom_drawing_in_2d.html
>
> But I don't know how to.

See https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#ref-role

Here you'd put a label before your section, as you can see done for the next section. We use `_doc_<name>_<section>` syntax by convention:

```
.. _doc_batching_batched_primitives

Batched primitives
^^^^^^^^^^^^^^^^^^
```

Then in the page where you want to add a link, you reference it with e.g.:
```
See :ref:`doc_batching_batched_primitives` for details on which primitives benefit from batching.
```

Note that the label takes a leading underscore, but then using it in a `:ref:` doesn't take it (don't ask me why :)).

The link title will be inferred from the section, so it will produce:

> See [Batched primitives](https://docs.godotengine.org/en/3.3/tutorials/optimization/batching.html#batched-primitives) for details on which primitives benefit from batching.

You can also customize the link title if need be with:
```
See :ref:`the batching reference <doc_batching_batched_primitives>` for details on which primitives benefit from batching.
```

**Comment by mhilbrunner** [comment]
Looks good - thank you :)

**Pull request merged**